source/windows/iocp/socket.c

```diff
@@ -357,8 +407,10 @@ static int s_socket_init(
         return AWS_OP_ERR;
     }
 
-    impl->vtable = &vtables[options->domain][options->type];
-    if (!impl->vtable || !impl->vtable->read) {
+    socket->vtable = &s_winsock_vtable;
+
+    impl->winsock_vtable = &s_winsock_vtables[options->domain][options->type];
```

Comment:
Can we make the AWS_ASSERT on domain a fatal assert instead?  Same with the assert on type.  Making the asserts fatal means we crash "safely" on invariant violation and not perform undefined behavior by reading from invalid static section memory.

Reply:
There is already assertion on domain and type: 
https://github.com/awslabs/aws-c-io/blob/6fcd0198047402687ce892cac07acc6c14493ceb/source/windows/iocp/socket.c#L401-L402

Reply:
I'm asking for those to become fatal.

Reply:
because https://github.com/awslabs/aws-c-common/blob/main/include/aws/common/assert.h#L66